packages/compose/index.ts

```diff
@@ -281,13 +281,13 @@ const mergeComposable: MergeComposable = (dstDescriptor, srcComposable) => {
 /**
  * TODO
  */
-const compose: ComposeMethod = function compose(this: unknown, ...args) {
+const compose: ComposeMethod = function compose(this: Stamp | unknown, ...args: (Composable | undefined)[]): Stamp {
```

Comment:
If we can remove `undefined` from here - it would be great.
If it's necessary to make `tsc` happy then let's leave it.

Reply:
I'll see what options we have

Reply:
I reverted compose/index.ts as @PopGoesTheWza suggested (makes total sense if we're going to be merging around) and then realised the code in this PR was probably written and tested using Typescript 4 before being merged into the PR branch. I'd say that is the cause of all the type strictness. Using Typescript 3 in the PR branch directly makes everything less strict and will allow us to start on the nasty merge sooner hehe.